# A restaurant offers apple and blueberry pies and stocks an equal number of each kind of pie. 

# Each day ten customers request pie. They choose, with equal probabilities, one of the two kinds of pie. 

# How many pieces of each kind of pie should the owner provide so that the probability is about .95 that each customer gets the pie of his or her own choice?

___

### Consider the shape of the binomial distribution:
![](images/11-1.png)

### What we want to know is: for what value on the x-axis is all the area under the curve to the right equal to 0.05?

![](images/11-2.png)

### We can calculate this by taking the sum of $b(10, 0.5, k)$ iterating from k=0 to k=10, but stopping once we cross 0.95

In [1]:
from math import factorial

In [2]:
def binom(n, p, k):
    numerator = factorial(n)
    denominator = factorial(k)*factorial(n-k)
    return float(numerator)/denominator*(p**k)*((1-p)**(n-k))

In [3]:
total = 0
for i in range(11):
    val = binom(10, 0.5, i)
    total += val
    print(i, total)
    if total >= 0.95:
        break

(0, 0.0009765625)
(1, 0.0107421875)
(2, 0.0546875)
(3, 0.171875)
(4, 0.376953125)
(5, 0.623046875)
(6, 0.828125)
(7, 0.9453125)
(8, 0.9892578125)


**At 8 of each kind of pie, we can be 98.9% confident that we won't run out**